services/web-app/components/page-nav/page-nav.js

```diff
@@ -0,0 +1,152 @@
+/*
+ * Copyright IBM Corp. 2021, 2022
+ *
+ * This source code is licensed under the Apache-2.0 license found in the
+ * LICENSE file in the root directory of this source tree.
+ */
+import { Column, Grid } from '@carbon/react'
+import clsx from 'clsx'
+import Link from 'next/link'
+import PropTypes from 'prop-types'
+import { useCallback, useEffect, useState } from 'react'
+
+import useEventListener from '@/utils/use-event-listener'
+import { mediaQueries, useMatchMedia } from '@/utils/use-match-media'
+
+import styles from './page-nav.module.scss'
+
+const PageNav = ({ contentRef, items = [] }) => {
+  const isLg = useMatchMedia(mediaQueries.lg)
+
+  const [lastActiveLink, setLastActiveLink] = useState(null)
+  const [activeItem, setActiveItem] = useState(null)
+
+  const handleRAF = () => {
+    if (!activeItem || (!checkIfSectionIdIsActive(activeItem) && lastActiveLink === activeItem)) {
+      window.requestAnimationFrame(setSelectedItem)
+    }
+  }
+
+  const handleHashChange = useCallback(() => {
+    const hash = window?.location?.hash?.replace('#', '') ?? null
+    setActiveItem(hash)
+    checkIfSectionIdIsActive(hash)
+  }, [])
+
+  useEventListener('scroll', handleRAF)
+  useEventListener('hashchange', handleHashChange)
+
+  useEffect(() => {
+    handleHashChange()
+  }, [handleHashChange])
+
+  const checkIfSectionIdIsActive = (id) => {
+    const section = document.getElementById(id)
```

Comment:
Do we want to search the entire document for the `id` or just what's inside the `contentRef` with a `contentRef.current.querySelectorAll()`. I don't think it really matters? But that might be safer?

Reply:
I could not get it to work using contentRef here